core/src/apps/cardano/sign_tx.py

```diff
@@ -502,27 +483,27 @@ def _cborize_shelley_witnesses(
     shelley_witnesses = []
 
     # include only one witness for each path
-    paths = set()
+    paths: List[List[int]] = list()
     for tx_input in inputs:
         if is_shelley_path(tx_input.address_n):
-            paths.add(tuple(tx_input.address_n))
+            append_unique(paths, tx_input.address_n)
```

Comment:
instead of having a helper I'd consider doing just:
```suggestion
            paths.append(tx_input.address_n) if tx_input.address_n not in paths
```
a bit repetitive but simple and explicit